my_project/tab_select/app_select.py

```diff
@@ -172,14 +173,45 @@ def submitted_data(
         except Exception as e:
             print(e)
             return (
-                None,
                 None,
                 True,
                 messages_alert["wrong_extension"],
                 "warning",
             )
     raise PreventUpdate
 
+#add switch_si_ip function and convert the data-store
+@app.callback(
+    [
+        ServersideOutput("df-store", "data"),
+        Output("si-ip-unit-store","data"),
+    ],
+    [
+        Input("si-ip-radio-input", "value"),
+        Input("meta-store", "data"),
+    ],
+    State("url-store", "data"),
+)
+
+def switch_si_ip(si_ip_input, meta, url_store):
+    si_ip_store = si_ip_input
+    lines = get_data(url_store)
```

Comment:
here you are requesting the data again and I guess this is slowing down the code significantly

Reply:
See other comments.